lib/components/TextInput.jsx

```diff
@@ -0,0 +1,203 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      isPristine: true,
+      isValid: false,
+      hasError: false,
+      errorMessage : null
+    };
+  }
+
+  /**
+   * If a value or errorMessage is passed as props, adjust the state accordingly
+   */
+  componentWillMount() {
+    if (this.props.errorMessage) {
+      this.setState({
+        isValid: false,
+        hasError: true,
+        errorMessage: this.props.errorMessage
+      });
+    }
+    if (this.props.value) {
+      this.setState({
+        value: this.props.value
+      });
+    }
+    if (this.props.isValid) {
+      this.setState({
+        isPristine: false,
+        isValid: true,
+        hasError: false,
+        errorMessage : null
+      });
+    }
+  }
+
+  /**
+   * @returns {node} (div) that includes a label, errorMessage, input 
+   */
+  render() {
+    let error = null;
+    if (this.state.hasError) {
+      error = (
+        <span className="usa-input-error-message" role="alert">
+          {this.state.errorMessage}
+        </span>
+      );
+    }
+
+    return (
+      <div className={this.state.hasError ? 'usa-input-error' : 'usa-input'}>
+        <InputLabel htmlFor={this.props.id} required={this.props.required} label={this.props.label} />
+
+        {error}
+
+        <input
+          id={this.props.id}
+          name={this.props.id}
+          type="text"
+          value={this.state.value}
+          className={this.state.isValid ? 'usa-input-success' : null}
+          required={this.props.required}
+          aria-required={this.props.required}
+          aria-labelledby={`${this.props.id}-label`}
+          onBlur={this._handleBlur.bind(this)}
+          onChange={this._handleChange.bind(this)}
+          spellCheck={this.props.spellCheck ? true : false}
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Checks input value against required status and passed validators
+   */
+  _validate() {
+    let validators = this.props.validators;
+    // Check if field empty
+    if (!this.state.value) {
+      // If it's required, say so
+      if (this.props.required) {
+        this.setState({
+          hasError: true,
+          isValid: false,
+          errorMessage: 'This field is required'
+        });
+      } else {
+        // is empty so reset isValid and hasError
+        this.setState({
+          isValid: false,
+          hasError: false,
+          errorMessage: null
+        });
+      }
+    }
+    // If validator(s) were sent as a prop, test them next
+    else if (validators) {
+      for (let validator of validators) {
+        // check is validator is forced 'no validation'
+        if (!validator.test) {
+          break;
+        }
+        // if validator doesn't pass it own test, set error and abort loop
+        if (!validator.isValid(this.state.value)) {
+          this.setState({
+            hasError: true,
+            isValid: false,
+            errorMessage: validator.errorMessage
```

Comment:
Can we rename `errorMessage` to just `message` or `explanation` or something?  Failing a validator isn't really an error. Just a thought.